controllers/statefulset.go

```diff
@@ -0,0 +1,316 @@
+			}
+		}
+	}
+	if cr.Spec.NodeSelector != nil {
+		sset.Spec.Template.Spec.NodeSelector = cr.Spec.NodeSelector
+		log.Info("Added Node Selector")
+	}
+	if err := ctrl.SetControllerReference(cr, sset, r.Scheme); err != nil {
+		log.Info("Cannot set statefulSet OwnerReference", err.Error())
+	}
+	if r.rgCustomization(cr) {
+		addRGConfigmap(sset)
+	}
+	return sset
+}
+
+func deleteRedisStatefulSet(client client.Client) error {
+	statefulset := &appv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      statefulSetName,
+			Namespace: namespace,
+		},
+	}
+	err := client.Delete(context.TODO(), statefulset)
+	if err != nil && !errors.IsNotFound(err) {
+		log.Error(err, "Failed to delete search redisgraph statefulset", "name", statefulSetName)
+		return err
+	}
+	time.Sleep(1 * time.Second) //Sleep for a minute to avoid quick update of statefulset
+	log.Info("StatefulSet deleted", "name", statefulSetName)
+	return nil
+}
+
+func isStatefulSetAvailable(kclient client.Client) bool {
+	//check if statefulset is present if not we can assume the pod is not running
+	found := &appv1.StatefulSet{}
+	err := kclient.Get(context.TODO(), types.NamespacedName{Name: statefulSetName, Namespace: namespace}, found)
+	if err != nil && errors.IsNotFound(err) {
+		return false
+	}
+	return true
+}
+
+func statefulSetNeedsUpdate(client client.Client, deployment *appv1.StatefulSet) bool {
+	found := &appv1.StatefulSet{}
+	err := client.Get(context.TODO(), types.NamespacedName{Name: statefulSetName, Namespace: namespace}, found)
+	if err != nil {
+		return true
+	} else {
+		if !reflect.DeepEqual(found.Spec, deployment.Spec) ||
+			!reflect.DeepEqual(found.GetObjectMeta(), deployment.GetObjectMeta()) {
+			log.Info("Volume source and/or metadata needs to be updated for redisgraph Statefulset")
+			return true
+		} else {
+			log.Info("No changes required for Statefulset")
+			return false
+		}
+	}
+}
+func updateRedisStatefulSet(client client.Client, deployment *appv1.StatefulSet) {
+	found := &appv1.StatefulSet{}
+	err := client.Get(context.TODO(), types.NamespacedName{Name: statefulSetName, Namespace: namespace}, found)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			log.Info("Statefulset not found. Creating Statefulset ...")
+			err = client.Create(context.TODO(), deployment)
+			if err != nil {
+				log.Error(err, "Failed to create Statefulset")
+				return
+			}
+			log.Info("Statefulset created successfully")
+			return
+		}
+		log.Error(err, "Failed to fetch Statefulset")
+		return
+	} else {
+		deployment.ObjectMeta.ResourceVersion = found.ObjectMeta.ResourceVersion
+		err = client.Update(context.TODO(), deployment)
+		if err != nil {
+			log.Error(err, "Failed to update Statefulset")
+			return
+		}
+		log.Info("Volume source and/or metadata updated for redisgraph Statefulset")
+		return
+	}
+
+}
+
+func (r *SearchOperatorReconciler) rgCustomization(cr *searchv1alpha1.SearchOperator) bool {
```

Comment:
Can we rename to `redisCustomization` for consistency?  (I know we use `rg` in many other places, but this file is using `Redis`)

Reply:
or `isRedisCustomization()` for those of us used to JS naming conventions.  (I could have looked at the return type.)